torchao/quantization/utils.py

```diff
@@ -402,11 +402,12 @@ def groupwise_affine_quantize_tensor_from_qparams(
     if TORCH_VERSION_AT_LEAST_2_5 and w.shape[-1] > 1:
         int_data_device_type = int_data.device.type
         # Move to cpu, until issue with MPS memory management of temporary tensors is resolved
-        if int_data_device_type == "mps":
-            int_data = int_data.cpu()
-        int_data = (int_data[::, ::2] << 4 | int_data[::, 1::2]).to(torch.uint8)
-        if int_data_device_type == "mps":
-            int_data = int_data.to(device="mps")
+        # if int_data_device_type == "mps":
+        #     int_data = int_data.cpu()
+        if int_data_device_type != "cpu":
+            int_data = (int_data[::, ::2] << 4 | int_data[::, 1::2]).to(torch.uint8)
+        # if int_data_device_type == "mps":
+        #     int_data = int_data.to(device="mps")
```

Comment:
please remove the code that's commented out

is this equivalent to previous code?

Reply:
According to https://github.com/pytorch/ao/pull/517#discussion_r1693296518, `<<` can be used in MPS backend, don't need to convert to `CPU` and use `CPU` backend. Since I don't have mps machine, I want to use CI to check if this can work. Otherwise, I can update to `int_data = (torch.bitwise_left_shift(int_data[::, ::2], 4) | int_data[::, 1::2]).to(torch.uint8)` instead.

Reply:
oh I see, makes sense